oxidation/crates/parsec_api_protocol/src/error.rs

```diff
@@ -0,0 +1,49 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use parsec_api_types::DateTime;
+use thiserror::Error;
+
+use crate::ApiVersion;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum HandshakeError {
+    #[error("Failed chalenge {0:?}")]
+    FailedChallenge(String),
+    #[error("Handshake bad administration token")]
+    BadAdministrationToken,
+    #[error("Handshake bad identity")]
+    BadIdentity,
+    #[error("Handshake organization expired")]
+    OrganizationExpired,
+    #[error("Handshake rvk mismatch")]
+    RVKMismatch,
+    #[error("Handshake revoked device")]
+    RevokedDevice,
+    #[error("Handshake out of ballpark {0:x?}")]
+    OutOfBallpark(ChallengeDataReport),
+    #[error("No overlap between client API versions {client_versions:?} and backend API versions {backend_versions:?}")]
+    APIVersion {
+        client_versions: Vec<ApiVersion>,
+        backend_versions: Vec<ApiVersion>,
+    },
+    #[error("{0}")]
```

Comment:
Can replace `String` by `&'static str`, and should have a `"Handshake invalid message: {0}"` error message

Reply:
I can't, because there is somewhere I use format! to build dynamic error

Reply:
The format was needed because of the `help` field that is deprecated (deprecated precisely because non-structured error value is hard to handle and easy to break ^^)

Without this `help` handling, `&'static str` is possible, and it simplify a lot error conversion, so much win \o/
(see 59ea8fa904b99158d4f4653a754a059dee085f17)